### PR TITLE
Update to latest egui master

### PR DESCRIPTION
**Pull request by emilk**
### What
With this update, we now format numbers in `DragValues` using `re_format`, i.e. with thousands separators. This makes large editable components easier to read, and makes our UI more consistent.

image

* Also closes https://github.com/rerun-io/rerun/issues/6681
* Also includes a better fix for https://github.com/rerun-io/rerun/pull/6639

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6683?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6683?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6683)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_space_view_time_series/src/space_view_class.rs [hunk 1]
```diff
@@ -277,13 +276,8 @@ impl SpaceViewClass for TimeSeriesSpaceView {
...
             ViewProperty::from_archetype::<ScalarAxis>(blueprint_db, ctx.blueprint_query, view_id);
-        let mut y_range = scalar_axis.component_or_fallback::<Range1D>(ctx, self, state)?;
-        if y_range.start() >= y_range.end() {
-            // Ensure that the range is valid - egui_plot might debug_assert otherwise.
-            // `next_up_f64` should be sufficient, but empirically it's not always enough
-            // (likely we're loosing precision due to some calculations down the line)
-            *y_range.end_mut() = y_range.start() + 1.0;
-        }
+        let y_range = scalar_axis.component_or_fallback::<Range1D>(ctx, self, state)?;
+        let y_range = make_range_sane(y_range);
```
Comment (Wumpf) [inline crates/re_space_view_time_series/src/space_view_class.rs:280]: nice. We have to do the same on the bar chart though I believe!

File: crates/re_ui/src/lib.rs [hunk 2]
```diff
@@ -89,6 +89,51 @@ pub fn apply_style_and_install_loaders(egui_ctx: &egui::Context) {
     );
 
     design_tokens().apply(egui_ctx);
+
+    egui_ctx.style_mut(|style| {
+        style.number_formatter = egui::style::NumberFormatter::new(format_with_decimals_in_range);
```
Comment (Wumpf) [inline crates/re_ui/src/lib.rs:94]: ohh, neat that this can be done like that now!

File: crates/re_ui/src/lib.rs [hunk 3]
```diff
@@ -89,6 +89,51 @@ pub fn apply_style_and_install_loaders(egui_ctx: &egui::Context) {
...
+    }
+
+    let epsilon = 16.0 * f32::EPSILON; // margin large enough to handle most peoples round-tripping needs
+
+    let min_decimals = *decimal_range.start();
+    let max_decimals = *decimal_range.end();
+    debug_assert!(min_decimals <= max_decimals);
+    debug_assert!(max_decimals < 100);
+    let max_decimals = max_decimals.min(16);
+    let min_decimals = min_decimals.min(max_decimals);
```
Comment (Wumpf) [inline crates/re_ui/src/lib.rs:116]: nit: the presence of `at_most`/`at_least` has made me weak :)
```suggestion
    let max_decimals = max_decimals.at_most(16);
    let min_decimals = min_decimals.at_most(max_decimals);
```

**Review by Wumpf — APPROVED** [review]
sweet!

File: crates/re_space_view_time_series/src/space_view_class.rs (hunk 1 above)
Comment (emilk) [reply crates/re_space_view_time_series/src/space_view_class.rs:280]: There is no `Range1D` in `crates/re_space_view_bar_chart/src/space_view_class.rs`

**Comment by Wumpf** [comment]
Actually I'm a bit torn on the looks for decimal separation after the dot 🤔 
image

File: crates/re_space_view_time_series/src/space_view_class.rs (hunk 1 above)
Comment (Wumpf) [reply crates/re_space_view_time_series/src/space_view_class.rs:280]: but @jleibs opened a pr specifically about the bar chart breaking under zero range

**Comment by emilk** [comment]
> Actually I'm a bit torn on the looks for decimal separation after the dot 🤔. Still useful but a less common sight

Yeah, that particular case doesn't look great. Maybe we should only add thousands separators after the dot if there is more than 5 or so digits after the dot. Still, this is a pre-existing problem in `re_format`. PRs welcome.

**Comment by emilk** [comment]
I can make a quick fix for `re_format`. Jeremy pointed out the same thing a few months back

**Pull request merged**